code/modules/projectiles/guns/energy/temperature.dm

```diff
@@ -1,14 +1,169 @@
 /obj/item/weapon/gun/energy/temperature
 	name = "temperature gun"
-	icon_state = "freezegun"
-	desc = "A gun that changes temperatures."
+	icon = 'icons/obj/gun_temperature.dmi'
+	icon_state = "tempgun_4"
+	item_state = "tempgun_4"
+	fire_sound = 'sound/weapons/pulse3.ogg'
+	desc = "A gun that changes the body temperature of its targets."
+	var/temperature = 300
+	var/target_temperature = 300
+	var/e_cost = 100
 	origin_tech = "combat=3;materials=4;powerstorage=3;magnets=2"
-	ammo_type = list(/obj/item/ammo_casing/energy/temp, /obj/item/ammo_casing/energy/temp/hot)
-	cell_type = "/obj/item/weapon/stock_parts/cell/high"
-	pin = null
 
+	ammo_type = list(/obj/item/ammo_casing/energy/temp)
+	cell_type = "/obj/item/weapon/stock_parts/cell"
+	var/powercost = ""
+	var/powercostcolor = ""
 
-/obj/item/weapon/gun/energy/temperature/attack_self(mob/living/user as mob)
-	select_fire(user)
+	var/emagged = 0			//ups the temperature cap from 500 to 1000, targets hit by beams over 500 Kelvin will burst into flames
+	var/dat = ""
+
+/obj/item/weapon/gun/energy/temperature/New()
+	..()
 	update_icon()
-	return
\ No newline at end of file
+	SSobj.processing |= src
+
+/obj/item/weapon/gun/energy/temperature/Destroy()
+	SSobj.processing -= src
+	return ..()
+
+/obj/item/weapon/gun/energy/temperature/newshot()
+	..()
+	chambered.temperature = temperature
+	chambered.e_cost = e_cost
+
+/obj/item/weapon/gun/energy/temperature/attack_self(mob/living/user as mob)
+	user.set_machine(src)
+	update_dat()
+	user << browse("<TITLE>Temperature Gun Configuration</TITLE><HR>[dat]", "window=tempgun;size=510x102")
+	onclose(user, "tempgun")
+
+/obj/item/weapon/gun/energy/temperature/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	if(istype(W, /obj/item/weapon/card/emag) && !emagged)
+		emagged = 1
+		user << "<span class='caution'>You double the gun's temperature cap ! Targets hit by searing beams will burst into flames !</span>"
+		desc = "A gun that changes the body temperature of its targets. Its temperature cap has been hacked"
+
+/obj/item/weapon/gun/energy/temperature/Topic(href, href_list)
+	if (..())
+		return
+	usr.set_machine(src)
+	src.add_fingerprint(usr)
+
+	if(href_list["temp"])
+		var/amount = text2num(href_list["temp"])
+		if(amount > 0)
+			src.target_temperature = min((500 + 500*emagged), src.target_temperature+amount)
```

Comment:
src may not be needed here.

I suggest you round the target temperature, especially when allowing direct input.